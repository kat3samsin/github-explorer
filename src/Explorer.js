import React, { Component } from 'react';
import {connect} from 'react-redux';

class Explorer extends Component {
  handleSubmit = (e) => {
    e.preventDefault();
    const searchKey = this.getSearch.value;
    const data = {
      id: new Date(),
      searchKey
    };

    this.props.dispatch({
      type: 'SHOW_PROJECTS',
      data
    });

  }
  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <input required type="text" ref={(input => this.getSearch = input)} placeholder="User or Organization Name" />&nbsp;
          <button>Search</button>
        </form>
      </div>
    );
  }
}
export default connect()(Explorer);