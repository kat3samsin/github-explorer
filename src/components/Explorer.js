import React, { Component } from 'react';
import {connect} from 'react-redux';
import { getProjects } from "../actions/getProjects";

export class Explorer extends Component {
  handleSubmit = (e) => {
    e.preventDefault();
    const searchKey = this.getSearch ? this.getSearch.value : '';
    const data = {
      id: new Date(),
      searchKey
    };

    this.props.search(data.searchKey);
  }
  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <input className='org' required type="text" ref={(input => this.getSearch = input)} placeholder="Enter organization name" /><br />
          <button className='button'>Search</button>
        </form>
      </div>
    );
  }
}
function mapDispatchToProps(dispatch) {
  return {
    search: (searchKey) => {
      dispatch(getProjects(searchKey))
   }
  }
 }
export default connect(null, mapDispatchToProps)(Explorer);