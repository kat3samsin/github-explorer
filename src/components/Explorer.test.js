import React from 'react';
import { shallow } from 'enzyme';
import { Explorer } from './Explorer';

describe('Explorer', () => {
  const searchFn = jest.fn();
  let wrapper;

   beforeEach(() => {
     wrapper = shallow(<Explorer search={searchFn} />)
   });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('when the form is submitted', () => {
    it('should call mock search', () => {
      wrapper.find('form').prop('onSubmit')({preventDefault() {}});
      expect(searchFn.mock.calls.length).toBe(1)
    });
  });
});