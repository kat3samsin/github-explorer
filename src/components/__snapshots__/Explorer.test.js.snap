// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explorer renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Explorer
    search={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <input
          className="org"
          placeholder="Enter organization name"
          required={true}
          type="text"
        />
        <br />
        <button
          className="button"
        >
          Search
        </button>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="org"
            placeholder="Enter organization name"
            required={true}
            type="text"
          />,
          <br />,
          <button
            className="button"
          >
            Search
          </button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "org",
            "placeholder": "Enter organization name",
            "required": true,
            "type": "text",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "button",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <input
            className="org"
            placeholder="Enter organization name"
            required={true}
            type="text"
          />
          <br />
          <button
            className="button"
          >
            Search
          </button>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="org"
              placeholder="Enter organization name"
              required={true}
              type="text"
            />,
            <br />,
            <button
              className="button"
            >
              Search
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "org",
              "placeholder": "Enter organization name",
              "required": true,
              "type": "text",
            },
            "ref": [Function],
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "button",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
